port/common/j9nls.c

```diff
@@ -385,6 +385,15 @@ j9nls_lookup_message(struct OMRPortLibrary *portLibrary, uintptr_t flags, uint32
 	omrthread_monitor_enter(nls->monitor);
 
 	if (!nls->catalog) {
+		/* Don't load the catalog if VM is running in an
+		 * english locale and there is a default message
+		 */
+		if (NULL != default_string) {
+			if (0 == strncmp(nls->language, "en", sizeof(nls->language))) {
```

Comment:
Looking at this again, I believe it is incorrect. If `nls->language` was `e` this check would report that they are the same string. An empty string for `nls->language` may also match but I have not verified that. `strncmp` is almost never the right answer as it allows for substring matches. Now if you want `e` and possible empty string to match then this is right otherwise it should be `strcmp`.

Reply:
Note, this is using `sizeof(nls->language)` to get a size of 4 based on the definition of the structure (which says `language` is a `char[4]`.

It's correct but can be rewritten as a strcmp if that prefer that